vp8_remb_simulcast.c

```diff
@@ -0,0 +1,571 @@
+        }
+		uint32_t current_bitrate = janus_vp8_remb_simulcast_get_bitrate_for_ramp_position(pRembContext, currentRampPos);
+		uint32_t higher_bitrate = 0;
+		if(nextHigherRampPos > currentRampPos)
+			higher_bitrate = janus_vp8_remb_simulcast_get_bitrate_for_ramp_position(pRembContext, nextHigherRampPos);
+
+		int lastLayerCounter = pRembContext->substream_switch_layer_counter;
+
+        // Start if we have a valid last bitrate we can use for calculations (2. remb package allows us to see wether the value goes up or down)
+        if(pSubscriber->last_bitrate_valid) {
+            double compareValue = dbBitrateChange;
+            if(pRembContext->last_bitrate_change > compareValue) {
+                compareValue = pRembContext->last_bitrate_change;
+            }
+
+    		/* We have layer bitrates so the remb simulcast switching shall be active */
+    		if(currentRampPos > 0 && bitrate < current_bitrate) {
+                /* We are above layer 0 and the bitrate is currently lower than we need for the current layer */
+    			if(bitrate < current_bitrate * 0.8 && compareValue < 0.2)	// If we are 20% below the lower bitrate and the bitself itself is not increasing by at least 0.2% between two remb messages
+    				pRembContext->substream_switch_layer_counter -= 4;
+    			else if(bitrate < current_bitrate * 0.9 && compareValue < 0.5) // If we are 10% below the lower bitrate and the bitself itself is not increasing by at least 0.5% between two remb messages
+    				pRembContext->substream_switch_layer_counter -= 2;
+    			else if(compareValue < 1) // If we are just below the lower bitrate and the bitrate is not growing by 1% between two remb messages
+    				pRembContext->substream_switch_layer_counter -= 1;
+    		}
+    		else if(bitrate > current_bitrate) {
+    			/* If the current bitrate is sufficient for the current layer increase the switcher value to 0 if it was negative before */
+    			if(pRembContext->substream_switch_layer_counter < 0)
+    				pRembContext->substream_switch_layer_counter ++;
+    			/* We are below the highest layer and the bitrate is currently higher than we need for the current layer */
+    			if(higher_bitrate && bitrate > higher_bitrate) {
+    				/* The current bitrate is higher than the next layer above */
+    				if(bitrate > higher_bitrate * 1.2)	// If we are 20% above the higher bitrate
+    					pRembContext->substream_switch_layer_counter += 4;
+    				else if(bitrate > higher_bitrate * 1.1) // If we are 10% above the higher bitrate
+    					pRembContext->substream_switch_layer_counter += 2;
+    				else  // If we are just above the higher bitrate
+    					pRembContext->substream_switch_layer_counter += 1;
+    			}
+    		}
+        }
+
+		double db_time_since_last_ramp_up = 0;
+		if(pRembContext->tm_last_ramp_up) {
+			db_time_since_last_ramp_up = (double)(g_get_monotonic_time() - pRembContext->tm_last_ramp_up) / 1000000;
```

Comment:
You can use `G_USEC_PER_SEC` instead of `1000000` which will ensure no risk of typos.